capsules/core/src/spi_controller.rs

```diff
@@ -186,8 +186,12 @@ impl<'a, S: SpiMasterDevice<'a>> Spi<'a, S> {
                 .take()
                 .unwrap_or((&mut [] as &'static mut [u8]).into());
             kwbuf.slice(0..write_len);
-            self.spi_master
-                .read_write_bytes(kwbuf, self.kernel_read.take())
+            if let Some(mut krbuf) = self.kernel_read.take() {
+                krbuf.slice(0..rlen);
```

Comment:
Do you not need a `cmp::min(kwbuf.len(), rlen)` here?

Reply:
I was copying line 178.

But..this does raise some questions about subslicemut...

Reply:
Fixed.